account.rs/src/wasm/src/poseidon.rs

```diff
@@ -0,0 +1,75 @@
+use ark_bn254::Fr;
+use js_sys::{Array, Uint8Array};
+use light_poseidon::{Poseidon, PoseidonBytesHasher, PoseidonError};
+use num_bigint::BigUint;
+use wasm_bindgen::{prelude::wasm_bindgen, JsValue};
+
+use crate::utils::set_panic_hook;
+
+#[wasm_bindgen]
+pub fn poseidon(inputs: &Array) -> Result<Uint8Array, JsValue> {
+    set_panic_hook();
+
+    let inputs_res: Result<Vec<Vec<u8>>, JsValue> = inputs
+        .iter()
+        .map(|val| {
+            if let Some(str_val) = val.as_string() {
+                let big_int = BigUint::parse_bytes(str_val.as_bytes(), 10)
+                    .ok_or_else(|| JsValue::from_str("Error parsing string to BigUint"))?;
+                Ok(big_int.to_bytes_be())
+            } else {
+                Err(JsValue::from_str(
+                    "All elements in the array should be strings representable as numbers",
+                ))
+            }
+        })
+        .collect();
+
+    let hash_res = poseidon_hash(inputs_res?);
+    match hash_res {
+        Ok(val) => {
+            let js_arr = Uint8Array::from(&val[..]);
+            Ok(js_arr)
+        }
+        Err(e) => Err(JsValue::from_str(&e.to_string())),
+    }
+}
+
+pub fn poseidon_hash(input: Vec<Vec<u8>>) -> Result<Vec<u8>, PoseidonError> {
+    let temp: Vec<&[u8]> = input.iter().map(AsRef::as_ref).collect();
+    let input_slice = temp.as_slice();
+
+    let hasher = Poseidon::<Fr>::new_circom(input.len());
```

Comment:
2 unrelated reasons:
1. It's just a unwrapping in the Swift-style, need to get used to Rust idioms :)
2. If we will unwrap the hasher on line 42, it will be necessary to us to make this variable mutable and I don't like it.